### PR TITLE
feat: Add encryption_key param to Blob.from_string

### LINKED ISSUE #390: Add encryption_key param to Blob.from_string
**Is your feature request related to a problem? Please describe.**
When I need to generate `Blob` object by using factory function `Blob.from_string`, I cannot use customer-supplied encryption option smoothly. I always need to assign internal variables like below:

```python
blob = Blob.from_string("gs://somewhere/sample.txt", client=client)
blob._encryption_key = encryption_key
```

 **Describe the solution you'd like**
* Add `encryption_key` parameter to `Blob.from_string`.

 **Describe alternatives you've considered**
Maybe adding explicit setter for `_encryption_key` is possible, but I think the solution above is more effective for resolving this issue.

 **Additional context**
I have a draft PR for this fix, so please review it if this request is acceptable. Thanks!

### PR DESCRIPTION
**Pull request by harrydrippin**
This PR contains several fixes as follows:

* Add `encryption_key` parameter to Blob.from_string.
* Fix several typos and format errors.

Fixes #390 🦕

### CODE REVIEW

File: tests/unit/test_blob.py [hunk 1]
```diff
@@ -4673,6 +4673,19 @@ def test_from_string_w_domain_name_bucket(self):
         self.assertEqual(blob.name, "b")
         self.assertEqual(blob.bucket.name, "buckets.example.com")
 
+    def test_from_string_w_encryption_key(self):
+        from google.cloud.storage.blob import Blob
+
+        connection = _Connection()
+        client = _Client(connection)
+        uri = "gs://bucket-name/blob-name"
+        encryption_key = "0123456789abcdef0123456789abcdef"
```
Comment (cojenco) [inline tests/unit/test_blob.py:4682]:
```suggestion
        encryption_key = b"0123456789abcdef0123456789abcdef"
```

File: google/cloud/storage/blob.py [hunk 2]
```diff
@@ -339,22 +339,27 @@ def public_url(self):
         )
 
     @classmethod
-    def from_string(cls, uri, client=None):
+    def from_string(cls, uri, client=None, encryption_key=None):
         """Get a constructor for blob object by URI.
 
          :type uri: str
```
Comment (cojenco) [inline google/cloud/storage/blob.py:345]: Thanks for catching this. For the doc string, the indentation should be left-indented instead and aligned with the triple quotes. Please help fix, thanks!

**Review by cojenco — CHANGES_REQUESTED** [review]
Thanks @harrydrippin for submitting a PR and helping us improve. Please refer to the comments for a few change requests. Feel free to let me know if any questions. Thank you!

**Review by tseaver — APPROVED** [review]

**Comment by tseaver** [comment]
@harrydrippin Thanks for your patch.  We've decided to use a getter/setter for encryption key: https://github.com/googleapis/python-storage/pull/409.

**Pull request closed without merging**